cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -494,4 +494,13 @@
     /{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-discard-ns rix,
     /var/lib/snapd/hostfs/{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-discard-ns rix,
 
+    # As a special exception, allow HOME to be /var/lib/jenkins
```

Comment:
For the future, the way I always imagined this would work for alternate actual home directories would be:
a) there is a `snap` command the user can run. Eg, `snap set-homes /var/nfs/users,/for/bar/homes`
b) snapd drops a file into `/etc/apparmor.d/tunables/home.d` to expand HOMEDIRS to include the above
c) snapd drops a file into /var/lib/snapd/apparmor/snap-confine to allow whatever apparmor policy is needed
d) snapd somehow communicates with snap-confine that it needs to add a bind mount for the above

In this manner, for normal systems where home is /home, nothing changes and systems where it isn't, it is configured in a clean manner.

Now, for the case of snaps that want to use an arbitrary home, we could expose something in snap.yaml for the above for snaps to declare, and via snap declaration/etc we allow it. However, we already have something very similar to this: layouts. If this was a strict mode snap, there would be no problem. The issue afaics is that because jenkins is classic, this breaks. Perhaps there is a way to expose layouts to classic snaps in a controlled manner... I don't have suggestions on that, just putting it out there...